Move CI runners away from `macOS-10.15` because of sunsetting by Github.

NlightNFotis opened the pull request:
There's a scheduled removal of the image by 30 Aug 2022, with intermittent
failing of those runners to raise awareness to that effect. We need to
move our CI runners to at least `macOS-11` to preempt any failures.

Fixes #7027.

NlightNFotis committed:
Move CI runners away from `macOS-10.15` because of sunsetting by Github.

There's a scheduled removal of the image by 30 Aug 2022, with intermittent
failing of those runners to raise awareness to that effect. We need to
move our CI runners to at least `macOS-11` to preempt any failures.

Fixes #7027.

tautschnig commented on .github/workflows/pull-request-checks.yaml:
```diff
@@ -382,8 +382,8 @@ jobs:
 
-  check-macos-10_15-make-clang:
-    runs-on: macos-10.15
+  check-macos-11-make-clang:
+    runs-on: macos-11
```
Should we then perhaps get rid of `check-macos-11-cmake-clang` and be content that we have a `make`-based test for macOS 11 and a `CMake`-based test for macOS 12?

NlightNFotis replied:
I diffed the two jobs, and they are essentially the same:

```diff
fotiskoutoulakis@192 ~ % diff /tmp/12.yml /tmp/11.yml 
1,2c1,2
< check-macos-12-cmake-clang:
<     runs-on: macos-12
---
> check-macos-11-cmake-clang:
>     runs-on: macos-11
```

So, the only difference I would reckon is the system toolchain (and other environment parameters), which if they are appropriately exercised by the `cmake` build, then I don't see any reason to keep it.

NlightNFotis replied:
Actually, I went ahead and did just that.

NlightNFotis committed:
Delete extraneous `check-macos-11-cmake-clang` job.

At this point, we have a `cmake` build on both macOS 11 and 12.

It makes more sense to drop one of them (`11`), given that the
environment and toolchain are exercised by the `make` job on the
same platform.

That allows us to simplify our CI pipeline.
